sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -111,6 +113,63 @@ object RapidsPluginUtils extends Logging {
     }
   }
 
+  private def detectMultipleJar(propName: String, jarName: String, conf: RapidsConf): Unit = {
+    val classloader = ShimLoader.getShimClassLoader()
+    val possibleRapidsJarURLs = classloader.getResources(propName).asScala.toSet.toSeq.filter {
+      url => {
+        val urlPath = url.toString
+        // filter out submodule jars and files stored under subdirs of '!/'
+        !urlPath.contains("rapids-4-spark-") && urlPath.contains("!/" + propName)
```

Comment:
Should this be only endsWith in reality?
```suggestion
        !urlPath.contains("rapids-4-spark-") && urlPath.endsWith("!/" + propName)
```